File: others/src/main/java/app/lawnicons/helper/CommonUtil.java
package app.lawnicons.helper;

import com.google.common.base.Charsets;
import org.apache.commons.io.FileUtils;
import org.apache.commons.io.FilenameUtils;
import org.dom4j.Document;
import org.dom4j.DocumentException;
import org.dom4j.Element;
import org.dom4j.io.OutputFormat;
import org.dom4j.io.SAXReader;
import org.dom4j.io.XMLWriter;

import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.nio.file.Path;
import java.util.List;

public class CommonUtil {

    public static List<Element> getElements(Document document, String path) throws DocumentException {
        Element rootElement = document.getRootElement();
        List<Element> list = rootElement.elements(path);
        return list;
    }

    public static Document getDocument(String xmlPath) throws DocumentException {
        SAXReader reader = new SAXReader();
        reader.setEncoding(Charsets.UTF_8.name());
        Document document = reader.read(xmlPath);
        return document;
    }

    public static String getFileWithXMlExtension(Path target, String extension, String extensionSuffix){
        String svgFilePath =  target.toFile().getAbsolutePath();
        StringBuilder svgBaseFile = new StringBuilder();
        int index = svgFilePath.lastIndexOf(".");
        if(index != -1){
            String subStr = svgFilePath.substring(0, index);
            svgBaseFile.append(subStr);
        }
        svgBaseFile.append(null != extensionSuffix ? extensionSuffix : "");
        svgBaseFile.append(".");
        svgBaseFile.append(extension);
        return svgBaseFile.toString();
    }
    public static void writeDocumentToFile(Document outDocument, String outputConfigPath) throws IOException {
        new File(FilenameUtils.getFullPath(outputConfigPath)).mkdirs();
        FileWriter fileWriter = new FileWriter(outputConfigPath);
        OutputFormat format = OutputFormat.createPrettyPrint();
        format.setEncoding(Charsets.UTF_8.name());
        XMLWriter writer = new XMLWriter(fileWriter, format);
        writer.write(outDocument);
        writer.close();
        fileWriter.close();
    }
}

File: others/src/main/kotlin/app/lawnchair/lawnicons/helper/CommonUtil.kt
package app.lawnchair.lawnicons.helper

import com.google.common.base.Charsets
import java.io.File
import java.io.FileWriter
import java.io.IOException
import java.nio.file.Path
import org.apache.commons.io.FilenameUtils
import org.dom4j.Document
import org.dom4j.DocumentException
import org.dom4j.Element
import org.dom4j.io.OutputFormat
import org.dom4j.io.SAXReader
import org.dom4j.io.XMLWriter

class CommonUtil {
    @Throws(DocumentException::class)
    fun getElements(
        document: Document,
        path: String?,
    ): List<Element?>? {
        val rootElement = document.rootElement
        return rootElement.elements(path)
    }

    fun sortedMapByValues(map: Map<String, String>): Map<String, String>? {
        val result2: MutableMap<String, String> = LinkedHashMap()
        map.entries.stream()
            .sorted(java.util.Map.Entry.comparingByValue())
            .forEachOrdered { (key, value): Map.Entry<String, String> ->
                result2[key] = value
            }
        return result2
    }

    @Throws(DocumentException::class)
    fun getDocument(xmlPath: String?): Document? {
        val reader = SAXReader()
        reader.encoding = Charsets.UTF_8.name()
        return reader.read(xmlPath)
    }

    fun getFileWithXMlExtension(
        target: Path,
        extension: String?,
        extensionSuffix: String?,
    ): String? {
        val svgFilePath = target.toFile().absolutePath
        val svgBaseFile = StringBuilder()
        val index = svgFilePath.lastIndexOf(".")
        if (index != -1) {
            val subStr = svgFilePath.substring(0, index)
            svgBaseFile.append(subStr)
        }
        svgBaseFile.append(extensionSuffix ?: "")
        svgBaseFile.append(".")
        svgBaseFile.append(extension)
        return svgBaseFile.toString()
    }

    @Throws(IOException::class)
    fun writeDocumentToFile(outDocument: Document?, outputConfigPath: String?) {
        File(FilenameUtils.getFullPath(outputConfigPath)).mkdirs()
        val fileWriter = FileWriter(outputConfigPath)
        val format = OutputFormat.createPrettyPrint()
        format.encoding = Charsets.UTF_8.name()
        val writer = XMLWriter(fileWriter, format)
        writer.write(outDocument)
        writer.close()
        fileWriter.close()
    }
}
